code/datums/gamemode/factions/bloodcult/bloodcult_runespells.dm

```diff
@@ -537,23 +590,437 @@
+
+
+/datum/rune_spell/conversion/Destroy()
+	if(conversion)
+		conversion.Die()
+	..()
+
+/datum/rune_spell/conversion/update_progbar()//progbar tracks conversion progress instead of paid blood
+	if (!progbar)
+		progbar = image("icon" = 'icons/effects/doafter_icon.dmi', "loc" = spell_holder, "icon_state" = "prog_bar_0")
+		progbar.pixel_z = WORLD_ICON_SIZE
+		progbar.plane = HUD_PLANE
+		progbar.layer = HUD_ABOVE_ITEM_LAYER
+		progbar.appearance_flags = RESET_COLOR
+	progbar.icon_state = "prog_bar_[min(100,round((100-remaining), 10))]"
+	return
+
+/datum/rune_spell/conversion/cast()
+	var/obj/effect/rune/R = spell_holder
+	R.one_pulse()
+	var/turf/T = R.loc
+	var/list/targets = list()
+
+	//first lets check for a victim above
+	for (var/mob/living/carbon/C in T)//all carbons can be converted...but only carbons. no cult silicons.
+		if (!iscultist(C))
+			//TODO: MOB NEEDS A MIND, leaving as is for now, so I can convert dummies to test stuff
+			targets.Add(C)
+	if (targets.len > 0)
+		victim = pick(targets)
+	else
+		to_chat(activator, "<span class='warning'>There needs to be a potential convert standing or lying on top of the rune.</span>")
+		qdel(src)
+		return
+
+	update_progbar()
+	if (activator.client)
+		activator.client.images |= progbar
+
+	//secondly, let's stun our victim and begin the ritual
+	to_chat(victim, "<span class='danger'>Occult energies surge from below your feet and seep into your body.</span>")
+	victim.Silent(5)
+	victim.Knockdown(5)
+	victim.Stun(5)
+	victim.overlay_fullscreen("conversionborder", /obj/abstract/screen/fullscreen/conversion_border)
+	victim.overlay_fullscreen("conversionred", /obj/abstract/screen/fullscreen/conversion_red)
+	victim.update_fullscreen_alpha("conversionred", 255, 5)
+	victim.update_fullscreen_alpha("conversionborder", 255, 5)
+	conversion = new(T)
+	flick("rune_convert_start",conversion)
+	playsound(R, 'sound/effects/convert_start.ogg', 75, 0, -4)
+
+	if (victim.mind)
+		if (victim.mind.assigned_role in impede_medium)
+			to_chat(victim, "<span class='warning'>Your sense of duty impedes down the ritual.</span>")
+			to_chat(activator, "<span class='warning'>Their will is strong, the ritual will take longer.</span>")
+
+		if (victim.mind.assigned_role in impede_hard)
+			to_chat(victim, "<span class='warning'>Your devotion to Nanotrasen impedes the ritual.</span>")
+			to_chat(activator, "<span class='warning'>Their willpower is amazing, the ritual will be exhausting.</span>")
+
+	for(var/obj/item/weapon/implant/I in victim)
+		if(I.implanted && istype(I,/obj/item/weapon/implant/loyalty))
+			to_chat(victim, "<span class='warning'>Your loyalty implants drastically slows down the ritual's progression.</span>")
+			to_chat(activator, "<span class='warning'>Their mind seems to reject the ritual by reflex. The ritual will take much longer.</span>")
+			break
+
+
+	spawn()
+		while (remaining > 0)
+			if (destroying_self || !spell_holder || !activator || !victim)
```

Comment:
`destroying_self` was not added in this PR but did you not know about `gcDestroyed`?

Reply:
isn't that one set to 1 only after qdel is called?

Reply:
I only saw that var being set in `Destroy`, so I assumed you just needed `gcDestroyed`. You can ignore this